### PR TITLE
DOC Document standardised CMSEditLink method

### PR DESCRIPTION
**Pull request by GuySartorelli**
Documents changes made in https://github.com/silverstripe/silverstripe-framework/pull/11338 and https://github.com/silverstripe/silverstripe-admin/pull/1809

## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/11258

### CODE REVIEW

File: en/02_Developer_Guides/15_Customising_the_Admin_Interface/01_ModelAdmin.md [hunk 1]
```diff
@@ -167,7 +167,7 @@ $tabLink = $admin->getLinkForModelTab('product-category');
 > [getModelTabForModelClass()](api:SilverStripe\Admin\ModelAdmin::getModelTabForModelClass()) method
 > for your `ModelAdmin` subclass.
 
-You can also use the new [CMSEditLinkExtension](api:SilverStripe\Admin\CMSEditLinkExtension) to provide a `CMSEditLink()` method on the record - see [Managing Records](../model/managing_records#getting-an-edit-link).
+You can also use the new [CMSEditLinkExtension](api:SilverStripe\Admin\CMSEditLinkExtension) to update the value of the `CMSEditLink()` method on the record - see [Managing Records](../model/managing_records#getting-an-edit-link).
```
Comment (emteknetnz) [inline en/02_Developer_Guides/15_Customising_the_Admin_Interface/01_ModelAdmin.md]: Should this be more along the links of "you can use the updateCMSEditLink() extension hook to update the value"?

Also CMSEditLinkExtension is no longer "new" so please remove that while you're there

File: en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md [hunk 2]
```diff
@@ -63,9 +64,10 @@ nesting `GridField`s. For the below examples it is assumed you aren't using nest
 will handle those situations for you if you use it.
 
 > [!TIP]
-> The easiest way to implement `CMSEditLink()` is by
+> The easiest way to get a correct return value for `CMSEditLink()` is by
```
Comment (emteknetnz) [inline en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md]: Should use the new hook instead?

File: en/02_Developer_Guides/00_Model/13_Managing_Records.md [hunk 3]
```diff
@@ -12,6 +12,8 @@ of some other record or directly [in a ModelAdmin](../customising_the_admin_inte
 
 ## Getting an edit link
 
+The [`DataObject::CMSEditLink()`](api:SilverStripe\ORM\DataObject::CMSEditLink()) method will give you the edit link if there is one, but by default it just returns `null`.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/00_Model/13_Managing_Records.md]:
```suggestion
The [`DataObject::getCMSEditLink()`](api:SilverStripe\ORM\DataObject::getCMSEditLink()) method will give you the edit link if there is one, but by default it just returns `null`.
```

File: en/02_Developer_Guides/14_Files/07_File_Usage.md [hunk 4]
```diff
@@ -59,7 +59,7 @@ class UsedOnTableExtension extends Extension
     // DataObject the File is used on.  This is useful for two reasons:
     // - It gives context more context, for instance if File is used on a Content block, it can be used to show the
     //   Page that Content Block is used on
-    // - The DataObject may not have a `CMSEditLink()` implementation, though the ancestor DataObject does.
+    // - The DataObject may not return a value from the `CMSEditLink()` method, though the ancestor DataObject does.
```
Comment (emteknetnz) [inline en/02_Developer_Guides/14_Files/07_File_Usage.md]: Why not? Now that the method is directly on DataObject you can just implement it on your subclass?

File: en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md [hunk 5]
```diff
@@ -263,7 +256,7 @@ class Product extends DataObject implements CMSPreviewable
 {
     // ...
 
-    public function CMSEditLink()
+    public function CMSEditLink(): ?string
```
Comment (emteknetnz) [inline en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md]:
```suggestion
    public function getCMSEditLink(): ?string
```

File: en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md [hunk 6]
```diff
@@ -592,10 +580,8 @@ class Product extends DataObject implements CMSPreviewable
 }
 ```
 
-The CMSEditLink doesn't matter so much for this implementation. It is required
-by the `CMSPreviewable` interface so some implementation must be provided, but
-you can safely return `null` or an empty string with no repercussions in this
-situation.
+The `CMSEditLink()` method doesn't matter so much for this implementation, so you can let
```
Comment (emteknetnz) [inline en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md]:
```suggestion
The `getCMSEditLink()` method doesn't matter so much for this implementation, so you can let
```

File: en/08_Changelogs/6.0.0.md [hunk 7]
```diff
@@ -185,6 +185,7 @@ SilverStripe\UserForms\Model\UserDefinedForm:
 
 - Native indexed PHP arrays can now be passed into templates and iterated over with `<% loop $MyArray %>`. Under the hood they are wrapped in [`ArrayList`](api:SilverStripe\View\ViewableData), so you can get the count using `$Count` and use `<% if $ArrayList %>` as a shortcut for `<% if $ArrayList.Count %>`. Other functionality from `ArrayList` such as filtering and sorting cannot be used on arrays since they don't have keys to filter or sort against.
 - Modules no longer need to have a root level `_config.php` or `_config` directory to be recognised as a Silverstripe CMS module. They will now be recognised as a module if they have a `composer.json` file with a `type` of `silverstripe-vendormodule` or `silverstripe-theme`.
+- A new [`DataObject::CMSEditLink()`](api:SilverStripe\ORM\DataObject::CMSEditLink()) method has been added, which returns `null` by default. This provides more consistency for that method which has previously been inconsistently applied to various subclasses of `DataObject`. See [managing records](/developer_guides/model/managing_records/#getting-an-edit-link) for more details about providing sane values for this method in your own subclasses.
```
Comment (emteknetnz) [inline en/08_Changelogs/6.0.0.md]:
```suggestion
- A new [`DataObject::getCMSEditLink()`](api:SilverStripe\ORM\DataObject::getCMSEditLink()) method has been added, which returns `null` by default. This provides more consistency for that method which has previously been inconsistently applied to various subclasses of `DataObject`. See [managing records](/developer_guides/model/managing_records/#getting-an-edit-link) for more details about providing sane values for this method in your own subclasses.
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md (hunk 2 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/15_Customising_the_Admin_Interface/04_Preview.md]: I'm not sure what you're asking

File: en/02_Developer_Guides/14_Files/07_File_Usage.md (hunk 4 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/14_Files/07_File_Usage.md]: The default implementation doesn't return a value. Yes you _can_ implement it on your subclass. But you don't have to.

File: en/02_Developer_Guides/15_Customising_the_Admin_Interface/01_ModelAdmin.md (hunk 1 above)
Comment (ChloeSartorelli) [reply en/02_Developer_Guides/15_Customising_the_Admin_Interface/01_ModelAdmin.md]: That could be stated separately if you want it to be. But that extension exists to do that for you and calling out here is intentional.

**Review by emteknetnz — APPROVED** [review]

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Merge conflict

**Comment by GuySartorelli** [comment]
Rebased

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**